documentation/sse_user_manual.md

```diff
@@ -0,0 +1,117 @@
+    return err
+}
+if err = secureIndex.Add(token, "keyword2", "id1"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword2", "id2"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword3", "id2"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword4", "id1"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword4", "id2"); err != nil {
+    return err
+}
+if err = secureIndex.Add(token, "keyword4", "id3"); err != nil {
+    return err
+}
+```
+
+A `Search` query can be used to identify which documents contain e.g. `keyword4`:
+
+```go
+IDs, err := secureIndex.Search(token, "keyword4")
+if err != nil {
+    return err
+}
+
+fmt.Println(IDs)
+```
+
+```go
+Out:    ["id1", "id2", "id3"]
+```
+
+A keyword can be deleted from a document, below `keyword4` is deleted from `id1`.
+```go
+err = secureIndex.Delete(token, "keyword4", "id1")
+if err != nil {
+    return err
+}
+```
+
+After the `Delete` query, the `Search` query will now return a different output:
+```go
+IDs, err = secureIndex.Search(token, "keyword4")
+if err != nil {
+    return err
+}
+
+fmt.Println(IDs)
+```
+
+```go
+Out:    ["id2", "id3"]
+```
+
+## Implementation
+
+In this section, some more technical details about the implementation are given.
+
+Given a keyword and an identifer for an `Add` query, a label is computed based on the keyword as well as on a *counter* (explained below), and an Identifier struct representing the identifier is created. The label is then mapped to the Identifier as shown below, and the "label -> Identifier" correlation is stored in the secure index. The Identifier is sealed before it is stored in order to avoid having plaintext keywords or identifiers outside of the D1 Library. Note that the keyword is used to seal the Identifier which means that the sealed Identifier can only be unsealed if the keyword is known.
+
+In storage:
+```go
+label(keyword, counter) -> sealed Identifier(keyword, identifier)
+```
+
+An Identifier struct (before seal) contains the identifier itself as well as a `NextCounter` as shown below. `NextCounter` is used to compute the next label based on the same keyword. As explained in the [SSE scheme](#sse-scheme) section, every time a specific keyword is mapped to a new identifier, a slightly different version of the same label is used, i.e. the next label maps the same keyword to another sealed Identifier. If the keyword has only been mapped to a single identifier, then its sealed Identifier's `NextCounter` is 0. Given a keyword for a `Search` query, all the identifiers that it maps to, i.e. all the identifiers that contain the given keyword, are then easily found by going through the chain of `NextCounter`'s and computing the corresponding label for each counter. The chain is illustrated below. It is ensured that the counter used to compute the first label in the chain is always known.
```

Comment:
I think "slightly different" is incorrect. It is actually indistinguishable from random.